dotnet/src/Connectors/Connectors.Memory.Weaviate/WeaviateVectorStoreRecordCollection.cs

```diff
@@ -0,0 +1,383 @@
+        typeof(DateTime?),
+        typeof(DateTimeOffset),
+        typeof(DateTimeOffset?),
+        typeof(Guid),
+        typeof(Guid?),
+        typeof(bool),
+        typeof(bool?)
+    ];
+
+    /// <summary>Default JSON serializer options.</summary>
+    private static readonly JsonSerializerOptions s_jsonSerializerOptions = new()
+    {
+        PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
+        DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull,
+        Converters =
+        {
+            new WeaviateDateTimeOffsetConverter(),
+            new WeaviateNullableDateTimeOffsetConverter()
+        }
+    };
+
+    /// <summary><see cref="HttpClient"/> that is used to interact with Weaviate API.</summary>
+    private readonly HttpClient _httpClient;
+
+    /// <summary>Optional configuration options for this class.</summary>
+    private readonly WeaviateVectorStoreRecordCollectionOptions<TRecord> _options;
+
+    /// <summary>A definition of the current storage model.</summary>
+    private readonly VectorStoreRecordDefinition _vectorStoreRecordDefinition;
+
+    /// <summary>A dictionary that maps from a property name to the storage name that should be used when serializing it to JSON for data and vector properties.</summary>
+    private readonly Dictionary<string, string> _storagePropertyNames;
+
+    /// <summary>The key property of the current storage model.</summary>
+    private readonly VectorStoreRecordKeyProperty _keyProperty;
+
+    /// <summary>The data properties of the current storage model.</summary>
+    private readonly List<VectorStoreRecordDataProperty> _dataProperties;
+
+    /// <summary>The vector properties of the current storage model.</summary>
+    private readonly List<VectorStoreRecordVectorProperty> _vectorProperties;
+
+    /// <summary>The mapper to use when mapping between the consumer data model and the Weaviate record.</summary>
+    private readonly IVectorStoreRecordMapper<TRecord, JsonNode> _mapper;
+
+    /// <inheritdoc />
+    public string CollectionName { get; }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="WeaviateVectorStoreRecordCollection{TRecord}"/> class.
+    /// </summary>
+    /// <param name="httpClient"><see cref="HttpClient"/> that is used to interact with Weaviate API.</param>
+    /// <param name="collectionName">The name of the collection that this <see cref="WeaviateVectorStoreRecordCollection{TRecord}"/> will access.</param>
+    /// <param name="options">Optional configuration options for this class.</param>
+    public WeaviateVectorStoreRecordCollection(
+        HttpClient httpClient,
+        string collectionName,
+        WeaviateVectorStoreRecordCollectionOptions<TRecord>? options = null)
+    {
+        // Verify.
+        Verify.NotNull(httpClient);
+        Verify.NotNullOrWhiteSpace(collectionName);
+
+        // Assign.
+        this._httpClient = httpClient;
+        this.CollectionName = collectionName;
```

Comment:
@dmytrostruk 
To align with Weaviate's naming conventions, it might be helpful to sanitize the collection name. Implemented in v1
[Documentation link](https://weaviate.io/developers/weaviate/manage-data/collections#class)

Reply:
@atiq-bs23 Thanks for your comment. I'm not sure we need to perform name sanitization on Semantic Kernel side. In my opinion, users could handle naming logic and follow Weaviate rules. A couple of reasons for that:
1. If user will provide one name, but collection will be silently created with another sanitized name, there could be some questions why the name was changed. Also, our sanitization rules may not work for all users, so they still be forced to implement their own sanitization logic if needed, so having this logic on our side won't fix the problem entirely. 
2. Naming conventions and rules in Weaviate can change over time, which means that we will have to update the logic every time there are some changes on Weaviate side. In this case, it won't allow users to proceed with this connector until we push new changes. On the other hand, if users will be responsible to provide final valid collection name, they will also have the ability to change it anytime on their side if there are some changes in Weaviate. So, it's more flexible.

In short, it's better to use the name as it's provided in constructor. If it's incorrect, user will receive an exception and will be able to update the name according to Weaviate naming rules.

Reply:
Thanks for clarify.